module/mempool/herocache/backdata/heropool/linkedlist_test.go

```diff
@@ -0,0 +1,88 @@
+package heropool
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// TestRemovingElementsFromList - first insures that list has been initialized correclty,
+// then removes elements and tests lists consistency.
+func TestRemovingElementsFromList(t *testing.T) {
+	limit_capacity := uint32(4)
+	number_of_entities := uint32(4)
+	t.Run(fmt.Sprintf("%d-limit-%d-entities", limit_capacity, number_of_entities), func(t *testing.T) {
+		withTestScenario(t, limit_capacity, number_of_entities, LRUEjection, []func(*testing.T, *Pool, []*unittest.MockEntity){
+			func(t *testing.T, pool *Pool, entities []*unittest.MockEntity) {
+				testRemovingElementsFromList(t, pool, entities, LRUEjection)
+			},
+		}...,
+		)
+	})
+}
+
+func checkList(t *testing.T, expectedList []EIndex, pool *Pool, list *state) {
```

Comment:
Functions like `checkList` and `testRemovingElementsFromList` could be renamed to something more descriptive. For instance, `checkList` could be renamed to `assertListStateMatchesExpected`. Additionally, the purpose of some parameters is not immediately obvious. It’s good practice to name your variables and parameters descriptively and provide `godoc` comments for clarification.

Reply:
understood, thank you for the comment.

Reply:
I removed this file, but out of curiosity which parameter doesn't have a descriptive name ?

by godoc comment you meant a single line comment that describes what this function does ?

Reply:
Please find a sample `godoc`-style documentation here: https://github.com/onflow/flow-go/blob/master/network/alsp/manager/manager.go

Reply:
thank you, I see, let's not close this thread. I'll got through my change and will try to fix.

Reply:
I think we can mark this as "resolved"